Shared/ForChildren/Stories/List/StoriesListViewModel.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  StoriesListViewModel.swift
+//  Movapp (iOS)
+//
+//  Created by Jakub Ruzicka on 12.11.2022.
+//
+
+import Combine
+import Foundation
+
+extension StoriesSectionItem {
+    static func mock() -> StoriesSectionItem {
+        StoriesSectionItem(title: "O perníkové chaloupce",
+                           subtitle: "Прянична хатинка",
+                           image: "data/stories/cervena-karkulka/thumbnail",
+                           duration: "3 min",
+                           slug: "pernikova-chaloupka")
+    }
+}
+
+struct StoriesSectionItem: Identifiable, Hashable {
+    var id: UUID { UUID() }
+
+    let title: String
+    let subtitle: String
+    let image: String
+    let duration: String
+    let slug: String
+}
+
+struct StoriesSection: Identifiable {
+    var id: UUID { UUID() }
+
+    let title: String
+    let stories: [StoriesSectionItem]
+}
+
+enum StoriesState {
+    case loading
+    case loaded(content: [StoriesSection])
+    case error(message: String)
+}
+
+protocol StoriesListViewModeling: ObservableObject {
+    var state: StoriesState { get }
+    var viewAppeared: PassthroughSubject<Void, Never> { get }
+}
+
+class StoriesListViewModel: StoriesListViewModeling {
+
+    @Published var state: StoriesState = .loading
+    let viewAppeared = PassthroughSubject<Void, Never>()
+
+    private var cancellables: [AnyCancellable] = []
+    private let storiesRepository = StoriesRepository()
```

Comment:
trochu bych tu zachoval DI? Jako to máme všude jinde. 

tj ForChildrenRootView dostává StoriesRepository a to se pak předává níže.